test/e2e-test/trait_update_test.go

```diff
@@ -0,0 +1,196 @@
+package controllers_test
```

Comment:
please try refactor it to be unit-test, thanks! Only necessary cases use e2e test, this could be a unit test like dependency

Reply:
got it.